### PR TITLE
Added ability to mix NLU and Stories in one file

### LINKED ISSUE #6566: Enable storing NLU, Strories, Rules in the same file
Part of #6475

- Make it possible to store `nlu`,`stories`, `rules` keys within one `YAML` file.
- Domain stays out of the picture at the moment, because of `responses` key it shares with `RasaYAMLReader`, this should be implemented separately later.

### PR DESCRIPTION
**Pull request by degiz**
Closes #6566 

**Proposed changes**:
- store all schemas under `rasa/utils/schemas`

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/schemas/domain.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    type: "seq"
+    sequence:
+    - type: "str"
+  actions:
+    type: "seq"
+    sequence:
+    - type: "str"
+      required: True
+  responses:
+    # see rasa/utils/schemas.yml
```
Comment (federicotdn) [inline rasa/utils/schemas/domain.yml:28]: The file at this path doesn't appear to exist, maybe this comment is outdated.

File: tests/core/training/story_reader/test_yaml_story_reader.py [hunk 2]
```diff
@@ -324,3 +324,15 @@ def test_is_not_test_story_file_without_test_prefix(tmp_path: Path):
...
+def test_read_mixed_training_data_file(default_domain: Domain):
+    training_data_file = "data/test_mixed_yaml_training_data/training_data.yml"
+
+    reader = YAMLStoryReader(default_domain)
+    yaml_content = rasa.utils.io.read_yaml_file(training_data_file)
+
+    with pytest.warns(None) as record:
+        reader.read_from_parsed_yaml(yaml_content)
+
+    assert not len(record)
```
Comment (federicotdn) [inline tests/core/training/story_reader/test_yaml_story_reader.py:338]:
```suggestion
    with pytest.warns(None) as record:
        reader.read_from_parsed_yaml(yaml_content)
        assert not len(record)
```

File: tests/nlu/training_data/formats/test_rasa_yaml.py [hunk 3]
```diff
@@ -356,3 +356,14 @@ def test_nlg_multimedia_load_dump_roundtrip():
...
+
+def test_read_mixed_training_data_file():
+    training_data_file = "data/test_mixed_yaml_training_data/training_data.yml"
+
+    reader = RasaYAMLReader()
+
+    with pytest.warns(None) as record:
+        reader.read(training_data_file)
+
+    assert not len(record)
```
Comment (federicotdn) [inline tests/nlu/training_data/formats/test_rasa_yaml.py:369]:
```suggestion
    with pytest.warns(None) as record:
        reader.read(training_data_file)
        assert not len(record)
```

**Review by federicotdn — APPROVED** [review]
Looks good! :+1: :+1:  I've only added some small comments.

File: rasa/constants.py [hunk 4]
```diff
@@ -26,8 +26,8 @@
 PACKAGE_NAME = "rasa"
 
 CONFIG_SCHEMA_FILE = "nlu/schemas/config.yml"
-DOMAIN_SCHEMA_FILE = "core/schemas/domain.yml"
-SCHEMA_UTILS_FILE = "utils/schemas.yml"
```
[the remark this answers is not in the document]
Comment (degiz) [reply rasa/constants.py:30]: yes!!

**Pull request merged**